tests/catch/unit/event/Unit_hipEventRecord.cc

```diff
@@ -28,64 +28,90 @@ THE SOFTWARE.
 #include <hip_test_common.hh>
 
 TEST_CASE("Unit_hipEventRecord") {
-    size_t N = 4 * 1024 * 1024;
-    unsigned threadsPerBlock = 256;
-    int iterations = 1;
+  constexpr size_t N = 4 * 1024 * 1024;
+  unsigned threadsPerBlock = 256;
+  int iterations = 1;
 
-    unsigned blocks = (N + threadsPerBlock - 1) / threadsPerBlock;
-    if (blocks > 1024) blocks = 1024;
-    if (blocks == 0) blocks = 1;
+  unsigned blocks = (N + threadsPerBlock - 1) / threadsPerBlock;
+  if (blocks > 1024) blocks = 1024;
+  if (blocks == 0) blocks = 1;
 
-    printf("N=%zu (A+B+C= %6.1f MB total) blocks=%u threadsPerBlock=%u iterations=%d\n", N,
-           ((double)3 * N * sizeof(float)) / 1024 / 1024, blocks, threadsPerBlock, iterations);
-    printf("iterations=%d\n", iterations);
+  constexpr size_t Nbytes = N * sizeof(float);
 
-    size_t Nbytes = N * sizeof(float);
+  float *A_h, *B_h, *C_h;
+  float *A_d, *B_d, *C_d;
+  HipTest::initArrays(&A_d, &B_d, &C_d, &A_h, &B_h, &C_h, N);
 
-    float *A_h, *B_h, *C_h;
-    float *A_d, *B_d, *C_d;
-    HipTest::initArrays(&A_d, &B_d, &C_d, &A_h, &B_h, &C_h, N);
+  enum TestType {
+    WithFlags_Default = hipEventDefault,
+    WithFlags_Blocking = hipEventBlockingSync,
+    WithFlags_DisableTiming = hipEventDisableTiming,
+#if HT_AMD
+    WithFlags_ReleaseToDevice = hipEventReleaseToDevice,
+    WithFlags_ReleaseToSystem = hipEventReleaseToSystem,
+#endif
+    WithoutFlags
+  };
 
-    hipEvent_t start, stop;
+  auto flags = GENERATE(WithFlags_Default, WithFlags_Blocking, WithFlags_DisableTiming,
+#if HT_AMD
```

Comment:
You cannot have a ```#if ... #endif``` block inside the ```GENERATE``` since the latter is a macro.

Reply:
@jatinx : Here is what I see when I build the tests:
```
HIP_PR-2722/hip/tests/catch/unit/event/Unit_hipEventRecord.cc:57:2: error: embedding a directive within macro arguments has undefined behavior [-Werror,-Wembedded-directive]
#if HT_AMD
 ^
HIP_PR-2722/hip/tests/catch/unit/event/Unit_hipEventRecord.cc:59:2: error: embedding a directive within macro arguments has undefined behavior [-Werror,-Wembedded-directive]
#endif
 ^
```

Reply:
I can see it too by adding a -Wembedded-directive explicitly.
Its not seen with -Wall and -Werror.

Will change it.
Thanks for letting me know.